<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TalesQtLevelEditorWidget</class>
 <widget class="QWidget" name="TalesQtLevelEditorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>667</width>
    <height>636</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="MetatilePickerLabel" name="metatilePickerLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>256</width>
         <height>256</height>
        </size>
       </property>
       <property name="mouseTracking">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>metatile picker</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QWidget" name="widget_5" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_12">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Preferred</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QWidget" name="widget_2" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <property name="sizeConstraint">
              <enum>QLayout::SetDefaultConstraint</enum>
             </property>
             <item>
              <widget class="QToolButton" name="metatilePaintToolButton">
               <property name="toolTip">
                <string>Tile pencil tool (P)</string>
               </property>
               <property name="text">
                <string>P</string>
               </property>
               <property name="icon">
                <iconset resource="resources.qrc">
                 <normaloff>:/resources/pencilicon.png</normaloff>:/resources/pencilicon.png</iconset>
               </property>
               <property name="shortcut">
                <string>P</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="metatileAreaCloneButton">
               <property name="toolTip">
                <string>Tile area clone tool (L)</string>
               </property>
               <property name="text">
                <string>G</string>
               </property>
               <property name="icon">
                <iconset resource="resources.qrc">
                 <normaloff>:/resources/selecticon.png</normaloff>:/resources/selecticon.png</iconset>
               </property>
               <property name="shortcut">
                <string>L</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="objectEditorButton">
               <property name="toolTip">
                <string>Object editor tool (O)</string>
               </property>
               <property name="text">
                <string>O</string>
               </property>
               <property name="icon">
                <iconset resource="resources.qrc">
                 <normaloff>:/resources/objecticon.png</normaloff>:/resources/objecticon.png</iconset>
               </property>
               <property name="shortcut">
                <string>O</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="spawnEditorButton">
               <property name="toolTip">
                <string>Spawn editor tool (K)</string>
               </property>
               <property name="text">
                <string>S</string>
               </property>
               <property name="icon">
                <iconset resource="resources.qrc">
                 <normaloff>:/resources/spawnicon.png</normaloff>:/resources/spawnicon.png</iconset>
               </property>
               <property name="shortcut">
                <string>K</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Preferred</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_3" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="warpsButton">
            <property name="text">
             <string>Warps...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="levelPropertiesButton">
            <property name="text">
             <string>Level properties...</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>1</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="ObjectListEditorWidget" name="objectListWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>250</width>
           <height>16777215</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QPushButton" name="objectAddWidget">
           <property name="text">
            <string>Add...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="objectRemoveWidget">
           <property name="text">
            <string>Remove</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="autofillAllButton">
           <property name="text">
            <string>Autofill all</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Map objects: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="mapNumObjectsLabel">
           <property name="text">
            <string>##/##</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="0" column="1">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="LevelEditorPreviewLabel" name="levelEditorLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="mouseTracking">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>level editor</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="LevelPreviewScrollBar" name="levelHorizontalScrollBar">
       <property name="singleStep">
        <number>16</number>
       </property>
       <property name="pageStep">
        <number>128</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="LevelPreviewScrollBar" name="levelVerticalScrollBar">
       <property name="singleStep">
        <number>16</number>
       </property>
       <property name="pageStep">
        <number>128</number>
       </property>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="LevelHeaderMapSelectorTreeWidget" name="mapSelectorWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>250</width>
       <height>150</height>
      </size>
     </property>
     <property name="headerHidden">
      <bool>true</bool>
     </property>
     <attribute name="headerVisible">
      <bool>false</bool>
     </attribute>
     <column>
      <property name="text">
       <string notr="true">1</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="1" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_11">
     <item>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QCheckBox" name="gridCheckBox">
           <property name="text">
            <string>Grid</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Zoom:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="zoomComboBox">
           <property name="currentIndex">
            <number>2</number>
           </property>
           <item>
            <property name="text">
             <string>25%</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>50%</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>100%</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>200%</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>300%</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>400%</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0" rowspan="4">
          <widget class="QWidget" name="widget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QRadioButton" name="levelVisualViewButton">
              <property name="text">
               <string>Visual</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="levelEffectViewButton">
              <property name="text">
               <string>Behavior</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="levelVertSolidityButton">
              <property name="text">
               <string>Vert. solidity</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="levelHorizSolidityButton">
              <property name="text">
               <string>Horiz. solidity</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="bgCheckBox">
           <property name="text">
            <string>BG</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="fgCheckBox">
           <property name="text">
            <string>FG</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="objectsCheckBox">
           <property name="text">
            <string>Objects</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="spawnsCheckBox">
           <property name="text">
            <string>Spawns</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QWidget" name="widget_4" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Camera boundaries</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="cameraBoundsShowBox">
            <property name="text">
             <string>Show</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Top</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="cameraBoundTopBox">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Bottom</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="cameraBoundBottomBox">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Left</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="cameraBoundLeftBox">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Right</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="cameraBoundRightBox">
             <property name="maximum">
              <number>65535</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
  <zorder></zorder>
  <zorder>mapSelectorWidget</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LevelEditorPreviewLabel</class>
   <extends>QLabel</extends>
   <header>leveleditorpreviewlabel.h</header>
  </customwidget>
  <customwidget>
   <class>LevelHeaderMapSelectorTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>levelheadermapselectortreewidget.h</header>
  </customwidget>
  <customwidget>
   <class>MetatilePickerLabel</class>
   <extends>QLabel</extends>
   <header>metatilepickerlabel.h</header>
  </customwidget>
  <customwidget>
   <class>LevelPreviewScrollBar</class>
   <extends>QScrollBar</extends>
   <header>levelpreviewscrollbar.h</header>
  </customwidget>
  <customwidget>
   <class>ObjectListEditorWidget</class>
   <extends>QListWidget</extends>
   <header>objectlisteditorwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
